### PR TITLE
Issue #14881: use empty string when sentence ending not found

### PR DESCRIPTION
**Pull request by rdiachenko**
Part of #14881 

This PR is an experiment to avoid using Optional introduced in https://github.com/checkstyle/checkstyle/pull/16020, and represent sentence ending with empty string to keep code simpler.

### CODE REVIEW

**Comment by rdiachenko** [comment]
Github, generate report for SummaryJavadoc/all-examples-in-one

**Comment by rdiachenko** [comment]
Github, generate report for SummaryJavadoc/all-examples-in-one

**Review by romani — APPROVED** [review]
It is good.
No regression, it is what we need.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java:655]: Truthfully, we are not finding the ending of the sentence. We are returning the whole sentence.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
Comment (rdiachenko) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java:655]: I don' t understand what you suggest to change here. Could you be more specific?

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java:655]: By name alone, finding the end of the sentence suggests to me we return the position of the end. In this case, we are returning the whole "sentence" if a proper period is found. The rest of the javadoc goes along with this at `. If a sentence ending period was found, returns the whole string ...`

It may not be easy to name this. To me a method name would be something like `getSentenceToPeriod` and update the javadoc to follow.

`empty if no ending was found.` => `empty if no sentence with a period was found.`

Comment (rdiachenko) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java]: done

Comment (rdiachenko) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java]: @rnveach I applied your suggestion. Could you resolve this converstion and unblock the PR?

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java]: I am not assigned, besides there is discussions in admin discord around the handling of the original contributor's pull request and concerns about process mismanagement and its impact on contributor morale.

Comment (rdiachenko) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java]: @rnveach could you explain please why you keep this PR re-assigning and still keeping it blocked? 
image

Considering I addressed your notes above, could you clearly explain what blocks this PR and why?

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java]: DIscussion in discord is still going on. You were added to the discord channel.

Plus I am not assigned, so there is no block.

**Comment by rdiachenko** [comment]
Moving this to draft because the intention was to experiment with a different approach and check validation / regression pass fine.

**Comment by rdiachenko** [comment]
Closing this in favour of https://github.com/checkstyle/checkstyle/pull/16020#pullrequestreview-2523996842

**Pull request closed without merging**
